packages/core/core/src/factories/apiFactory/types.ts

```diff
@@ -0,0 +1,51 @@
+import { IntegrationPlugin } from './../../utils/nuxt';
+export interface ApiExtensionInstance {
+  beforeSetup?: (config, headers?: Record<string, string>) => any;
+  afterSetup?: ({ config, client }, headers?: Record<string, string>) => { config; client };
+  beforeCall?: ({ config, functionName, params }) => any;
+  afterCall?: ({ config, functionName, params }) => any;
+}
+
+export type ApiExtension = (req: any, res: any) => ApiExtensionInstance;
+
+export interface BaseApiFactoryParams<T, F = any> {
+  tag: string;
+  api: F;
+  isProxy?: boolean;
+}
+
+export interface ApiFactoryParams<T, F = any> extends BaseApiFactoryParams<T, F> {
```

Comment:
isn't that only for ApiClientFactory (so no proxy)? I assume ApiFactoryParams means Api(Client|Proxy)FactoryParams